main.tf

```diff
@@ -22,6 +22,17 @@ resource "aws_s3_bucket" "this" {
       }
     }
   }
+
+  dynamic "grant" {
+    for_each = try(length(var.grants), 0) == 0 || try(length(var.acl), 0) > 0 ? [] : var.grants
```

Comment:
Seems like the try function is overkill

Reply:
Given that grants is a list not sure why you would need the try function to evaluate the length.

Reply:
I would think you can just test acl by itself. It is its own true/false condition. It either contains something and is true or it is set to null and therefore is false.

Reply:
gotcha. tests worked (w/ and w/o grants) using the for_each we discussed in Slack.